website/content/docs/concepts/node-pools.mdx

```diff
@@ -0,0 +1,366 @@
+---
+layout: docs
+page_title: Node Pools
+description: Learn about the internal architecture of Nomad.
+---
+
+# Node Pools
+
+Node pools are a way to group clients and segment infrastructure into logical
+units that can be targeted by jobs for a strong control over where allocations
+are placed.
+
+Without node pools, allocations for a job can be placed in any eligible client
+in the cluster. Affinities and constraints can help express preferences for
+certain nodes, but they do not easily prevent other jobs from placing
+allocations in a set of nodes.
+
+A node pool can be created using the [`nomad node pool apply`][cli_np_apply]
+command and passing a node pool [specification file][np_spec].
+
+```hcl
+# dev-pool.nomad.hcl
+node_pool "dev" {
+  description = "Nodes for the development environment."
+
+  meta {
+    environment = "dev"
+    owner       = "sre"
+  }
+}
+```
+
+```shell-session
+$ nomad node pool apply dev-pool.nomad.hcl
+Successfully applied node pool "dev"!
+```
+
+Clients can then be added to this node pool by setting the
+[`node_pool`][client_np] attribute in their configuration file, or using the
+equivalent [`-node-pool`][cli_agent_np] command line flag.
+
+```hcl
+client {
+  # ...
+  node_pool = "dev"
+  # ...
+}
+```
+
+To help streamline this process, nodes can create node pools on demand. If a
+client configuration references a node pool that does not exist yet, Nomad
+creates the node pool automatically on client registration.
```

Comment:
```suggestion
To help streamline this process, nodes can create node pools on demand. If a
client configuration references a node pool that does not exist yet, Nomad
creates the node pool automatically on client registration. (However, see
[Multi-region Clusters](#multi-region-clusters) for an exception.)
```

Reply:
Good call. I actually forgot about this until I wrote the multi-region section but then forgot to come back here 😅